### PR TITLE
Add detailed description of pipeline configuration

### PR DESCRIPTION
**Pull request by djaglowski**
## **Please check that the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Add a changelog entry (for non-trivial bug fixes / features)
- [ ] CI passes

### CODE REVIEW

File: docs/pipeline.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
+# Pipeline
+
+A pipeline is made up of [operators](/docs/operators/README.md). The pipeline defines how stanza should input, process, and output logs. 
+
+
+## Liner Pipelines
```
Comment (camdencheek) [inline docs/pipeline.md:6]: `Liner` -> `Linear`

File: docs/pipeline.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+pipeline:
+  # Read and parse a JSON file
+  - type: file_input
+    includes: 
+      - my-log.json
+  - type: json_parser
+    output: stdout # flow directly to stdout
+  
+  # Read and parse a text file
+  - type: file_input
```
Comment (camdencheek) [inline docs/pipeline.md:118]: This pipeline is invalid because the two file inputs don't have `id`s

File: docs/pipeline.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+pipeline:
+  # Read log file
+  - type: file_input
+    includes: 
+      - my-log.txt
+
+  # Route based on log type
+  - type: router
+    routes:
+      - expr: '$.looks_like == "format_one"'
```
Comment (camdencheek) [inline docs/pipeline.md:142]: This is a little confusing because `$.looks_like` will never exist coming directly from a file input

Comment (camdencheek) [reply docs/pipeline.md:142]: Might be better to do something like `- expr: '$record startsWith "prefix1"'`

**Review by camdencheek — APPROVED** [review]
Just a couple of small comments, but this looks really great. It's a long-needed improvement to our docs

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|1.9828026|+0.18966353|127.01522|-3.7613144|
|1|5000|6.189956|-0.18951845|137.74057|+0.15167236|
|1|10000|12.224422|+0.12072945|145.83324|+1.9089355|
|1|50000|59.27614|+2.8224106|178.87608|+2.205017|
|1|100000|113.547615|+0.57702637|233.22899|-3.3863068|
|10|100|2.3621764|+0.120699644|134.68036|+1.1219025|
|10|500|7.224537|+0.41407013|137.36449|-0.9878845|
|10|1000|13.966119|+1.0176115|145.88025|-0.17524719|
|10|5000|63.518784|+0.7412033|184.13658|+2.693161|
|10|10000|122.950005|-4.294899|232.25377|-1.0098419|

**Pull request merged**